### ISSUE TITLE
[New Feature] Double click to equip item

### ISSUE THREAD

**Issue by Arutosio** [modernuo/ModernUO#458]
I think it could be very useful to implement this function: by double clicking on a weapon / armor / dress, it is equipped, in case you already have something equipped it is replaced with the one equipped.

**Comment by kamronbatman** [comment]
I think this would be era specific. Should we create a setting for this?
I am wondering if we should have this in modernuo.json, or somewhere else.
Thoughts?

**Comment by Arutosio** [comment]
> I think this would be era specific. Should we create a setting for this?
> 
> I am wondering if we should have this in modernuo.json, or somewhere else.
> 
> Thoughts?

in my opinion, you can add the setting on moderuo.json
"DoubleClickToEquip" : "true"

this setting in my opinion should be set to true by default

note: the behavior of objects that already have a function with the double click must not change with this, they must remain as it is

**Comment by karolswdev** [comment]
> > I think this would be era specific. Should we create a setting for this?
> > I am wondering if we should have this in modernuo.json, or somewhere else.
> > Thoughts?
> 
> in my opinion, you can add the setting on moderuo.json
> "DoubleClickToEquip" : "true"
> 
> this setting in my opinion should be set to true by default
> 
> note: the behavior of objects that already have a function with the double click must not change with this, they must remain as it is

What happens with objects that are both equippable and have a use script? What should be the order of execution in that case?

**Comment by Arutosio** [comment]
I don't understand, can you rephrase the question?

**Comment by kamronbatman** [comment]
@Arutosio, @karolswdev is asking about an item such as a torch or candle. It already has a double click.

I think for those, it equips then executes. Like Sphere.

**Comment by Arutosio** [comment]
> @Arutosio, @karolswdev is asking about an item such as a torch or candle. It already has a double click.
> 
> I think for those, it equips then executes. Like Sphere.

yes, like sphere

**Comment by Arthrutus** [comment]
This has never been a feature for weapons or armors on OSI. This is a purely custom feature which can be implemented via BaseArmor/BaseWeapon.cs (OnDoubleClick).

**Comment by kamronbatman** [comment]
I think we should have code for this in the MUO Scripts repo, or in a fork. We can pin it and let people know on Discord.
Note: This is not as easy to implement as people may think. Lots of random gotcha's!
